libcst/matchers/tests/test_findall.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# pyre-strict
+
+from textwrap import dedent
+from typing import Optional, Sequence
+
+import libcst as cst
+import libcst.matchers as m
+import libcst.metadata as meta
+from libcst.matchers import findall
+from libcst.testing.utils import UnitTest
+
+
+class MatchersFindAllTest(UnitTest):
```

Comment:
The test cases here test the deep_equals instead of identity. Can we also have some test cases to test the identity of nodes to make sure we get the nodes from the original tree?

Reply:
We already have tests for various pieces of the system, such as transform/visit. We shouldn't retest every part of the system for every new PR otherwise our tests will be unwieldy. We need to trust that a visitor that gathers indeed keeps nodes by identity as that's how its defined.

I don't think that adding even more testing here benefits us.

Reply:
I was thinking a test case like this.

```
a = 1
b = a + 2
```
When findall with ExpressionContext == STORE, we expect to get the first `Name("a")` instead of second `Name("a")`.

Reply:
That's how `ExpressionContext` works along with traversal, and we have tests for each piece of the system.